src/Microsoft.Framework.DesignTimeHost/ProtocolManager.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Framework.DesignTimeHost.Models;
+using Microsoft.Framework.Runtime;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Framework.DesignTimeHost
+{
+    public class ProtocolManager
+    {
+        /// <summary>
+        /// Environment variable for overriding protocol.
+        /// </summary>
+        public const string EnvDthProtocol = "DTH_PROTOCOL";
+
+        /// <summary>
+        /// Type name of the message representing the protocol negotiation.
+        /// </summary>
+        public const string NegotiationMessageTypeName = "ProtocolVersion";
+
+        public ProtocolManager(int capVersion)
```

Comment:
`highestVersion`? or just `version`? This is the highest version this runtime supports.